// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message IapBetaBrand {
  string application_title = 1;
  string name = 2;
  bool org_internal_only = 3;
  string support_email = 4;
  string project = 5;
}

message ApplyIapBetaBrandRequest {
  IapBetaBrand resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIapBetaBrandRequest {
  string service_account_file = 1;
  IapBetaBrand resource = 2;
}

message ListIapBetaBrandRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListIapBetaBrandResponse {
  repeated IapBetaBrand items = 1;
}

service IapBetaBrandService {
  rpc ApplyIapBetaBrand(ApplyIapBetaBrandRequest) returns (IapBetaBrand);
  rpc DeleteIapBetaBrand(DeleteIapBetaBrandRequest) returns (google.protobuf.Empty);
  rpc ListIapBetaBrand(ListIapBetaBrandRequest) returns (ListIapBetaBrandResponse);
}
